uint16 msg_seq
int16[] x
int16[] y